frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/NetworkAttachmentDefinitionDetails.tsx

```diff
@@ -0,0 +1,118 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import {
+  Firehose,
+  ScrollToTopOnMount,
+  SectionHeading,
+  StatusBox,
+} from '@console/internal/components/utils';
+import { getName, getNamespace } from '@console/shared/src';
+import { NetworkAttachmentDefinitionKind } from '../../types';
+import { getConfigAsJSON, getDescription, getType } from '../../selectors';
+import { networkTypes } from '../../constants';
+
+const NET_ATTACH_DEF_OVERVIEW_HEADING = 'Network Attachment Definition Overview';
+
+export const prefixedID = (idPrefix: string, id: string) =>
+  idPrefix && id ? `${idPrefix}-${id}` : null;
+
+export const DetailsItem: React.FC<DetailsItemProps> = ({
+  title,
+  idValue,
+  isNotAvail = false,
+  valueClassName,
+  children,
+}) => {
+  return (
+    <>
+      <dt>{title}</dt>
+      <dd id={idValue} className={valueClassName}>
+        {isNotAvail ? <span className="text-secondary">Not available</span> : children}
+      </dd>
+    </>
+  );
+};
+
+export const NetAttachDefinitionSummary: React.FC<NetAttachDefinitionSummaryProps> = ({
+  netAttachDef,
+}) => {
+  const name = getName(netAttachDef);
+  const description = getDescription(netAttachDef);
+  const type = getType(getConfigAsJSON(netAttachDef));
+  const id = `${getNamespace(netAttachDef)}-${getName(netAttachDef)}`;
+
+  return (
+    <React.Fragment>
+      <DetailsItem title="Name" idValue={prefixedID(id, 'name')} isNotAvail={!name}>
+        {name}
+      </DetailsItem>
+
+      <DetailsItem
+        title="Description"
+        idValue={prefixedID(id, 'description')}
+        valueClassName="kubevirt-vm-resource-summary__description"
+        isNotAvail={!description}
+      >
+        {description}
+      </DetailsItem>
+
+      <DetailsItem title="Type" idValue={prefixedID(id, 'type')} isNotAvail={!type}>
+        {_.get(networkTypes, [type], null) || type}
+      </DetailsItem>
+    </React.Fragment>
+  );
+};
+
+export const NetworkAttachmentDefinitionDetails: React.FC<NetAttachDefTabProps> = ({
+  obj: netAttachDef,
+}) => {
+  return (
+    <div className="co-m-pane__body">
+      <Firehose resources={[]}>
+        <NetworkAttachmentDefinitionDetailsBase netAttachDef={netAttachDef} />
+      </Firehose>
+    </div>
+  );
+};
+
+export const NetworkAttachmentDefinitionDetailsBase: React.FC<NetAttachDefDetailsProps> = (
+  props,
+) => {
+  const { netAttachDef, ...restProps } = props;
```

Comment:
`restProps` are not defined in `NetAttachDefDetailsProps`, can you please improve the type ?

Reply:
I was able to just remove `restProps` altogether.